embedded-service/src/ipc/rpc.rs

```diff
@@ -0,0 +1,83 @@
+//! Definitions for IPC
+use core::sync::atomic::AtomicBool;
+
+use embassy_sync::{blocking_mutex::raw::RawMutex, channel::Channel, mutex::Mutex};
+
+/// A simple RPC channel for executing commands
```

Comment:
bikeshedding: "RPC" makes me think it's going off-device when this is really just IPC. Another name could be simply "message handler"